packages/clerk-js/src/ui/signUp/SignUpStart.tsx

```diff
@@ -252,7 +258,7 @@ function _SignUpStart(): JSX.Element {
   ) : null;
 
   const shouldShowEmailAddressField =
```

Comment:
🙃 
At a later point or now where we have the feature fresh, I would extract this check to another file or at the end of this one and comment on the cases which this extensive boolean check applies to.